hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionImplementation.java

```diff
@@ -1125,10 +1130,21 @@ rpcControllerFactory, getMetaLookupPool(), connectionConfig.getMetaReadRpcTimeou
 
   void takeUserRegionLock() throws IOException {
     try {
+      long waitStartTime = 0;
       long waitTime = connectionConfig.getMetaOperationTimeout();
+      if (metrics != null) {
```

Comment:
Whenever we want to update the metrics, we always have to do a null check and it makes code too verbose.
We assign metrics to null if CLIENT_SIDE_METRICS_ENABLED_KEY is set to false. Code [here](https://github.com/apache/hbase/blob/branch-2/hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionImplementation.java#L341)
Instead of assigning to null, can we assign to some DUMMY_METRICS or EMPTY_METRICS and void all the methods. While populating the metrics, check whether metrics is an instance of DUMMY_METRICS. If yes then do nothing. This will make code look lot cleaner. WDYT? @vli02  @d-c-manning @apurtell

Reply:
I think this `if` should be ok and it is in a lot of places, it's not worth to make the changes. @d-c-manning @apurtell Your opinion is welcomed if you have any. Thanks!

Reply:
Created [HBASE-27697](https://issues.apache.org/jira/browse/HBASE-27697).